core/sup_tree_core/executor_pool/executor_pool.ex

```diff
@@ -147,7 +147,7 @@ defmodule AntikytheraCore.ExecutorPool do
     end
 
     defun start_link([]) :: {:ok, pid} do
-      Supervisor.start_link([AntikytheraCore.ExecutorPool], [strategy: :simple_one_for_one, name: __MODULE__])
+      DynamicSupervisor.start_link([strategy: :one_for_one, name: __MODULE__])
```

Comment:
Let's make this a module-based `DynamicSupervisor`.
- add `use DynamicSupervisor` and `init/1`
- remove `child_spec/1`